### PR TITLE
Fix Alluxio Yarn integration

### PR DESCRIPTION
**Pull request by cheng-chang**
1. There are classes in alluxio-core-common that depend on grpc, grpc is only included in alluxio-core-transport, so it has to be included, otherwise, ClassNotFound issues will happen. Previously, we exclude alluxio-core-protobuf because of some shading hacks which no longer work for alluxio-core-transport.
2. Alluxio uses a newer version of Guava, an older version of Guava also exists in the YARN_APPLICATION_CLASSPATH, so without this change, the classloader will load the older version of Guava, so MethodNotFound exception might be thrown in some Alluxio  classes due to their needs for the newer version of Guava.

### CODE REVIEW

File: integration/yarn/src/main/java/alluxio/yarn/Client.java [hunk 1]
```diff
@@ -394,13 +394,13 @@ private void setupContainerLaunchContext() throws IOException, YarnException {
 
   private void setupAppMasterEnv(Map<String, String> appMasterEnv) throws IOException {
     String classpath = ApplicationConstants.Environment.CLASSPATH.name();
+    Apps.addToEnvironment(appMasterEnv, classpath, PathUtils.concatPath(Environment.PWD.$(), "*"),
+        ApplicationConstants.CLASS_PATH_SEPARATOR);
```
Comment (calvinjia) [inline integration/yarn/src/main/java/alluxio/yarn/Client.java]: Could you add a comment to make sure we know the ordering of this matters?

File: integration/yarn/src/main/java/alluxio/yarn/Client.java [hunk 2]
```diff
@@ -394,13 +394,13 @@ private void setupContainerLaunchContext() throws IOException, YarnException {
 
   private void setupAppMasterEnv(Map<String, String> appMasterEnv) throws IOException {
     String classpath = ApplicationConstants.Environment.CLASSPATH.name();
+    Apps.addToEnvironment(appMasterEnv, classpath, PathUtils.concatPath(Environment.PWD.$(), "*"),
```
Comment (aaudiber) [inline integration/yarn/src/main/java/alluxio/yarn/Client.java]: can we add a comment on why we order it like this?

File: integration/yarn/src/main/java/alluxio/yarn/Client.java (hunk 1 above)
Comment (aaudiber) [reply integration/yarn/src/main/java/alluxio/yarn/Client.java]: jinx

**Comment by calvinjia** [comment]
@aaudiber We shade for the runtime client which is generally the artifact that is most prone to jar hell. We could consider shading other jars.

**Comment by cheng-chang** [comment]
alluxio-bot, check this please

**Comment by cheng-chang** [comment]
alluxio-bot, check this please

**Comment by cheng-chang** [comment]
alluxio-bot, check this please

**Comment by cheng-chang** [comment]
@calvinjia @aaudiber PTAL

**Comment by gpang** [comment]
@cheng-chang Thanks for this fix! Could you add some of the descriptions to the PR description? When we squash-and-merge, the PR title and description will translate into the commit message.

**Pull request merged**